Service/Endpoint/EndpointFactory.php

```diff
@@ -144,6 +146,35 @@ private function setEndpointResponseClass(Endpoint $endpoint, $responseClass = n
         return $endpoint->setResponseClass($responseClass);
     }
 
+    /**
+     * @param Endpoint    $endpoint
+     * @param string|null $errorClass
+     *
+     * @return EndpointInterface
+     *
+     * @throws ConfigurationException
+     */
+    private function setEndpointErrorClass(Endpoint $endpoint, $errorClass = null): EndpointInterface
+    {
+        if (null === $errorClass) {
+            return $endpoint;
+        }
+
+        $className = $errorClass;
+        if (preg_match('/.*\[\]$/', $className)) {
+            $className = substr($className, 0, -2);
+        }
```

Comment:
I think it was introduced to avoid a situation when the configuration is put as an array in YML. I would leave it as it is to keep the things the same